src/components/minimap/index.js

```diff
@@ -0,0 +1,422 @@
+    );
+  }
+
+  /**
+   * Remove window event listeners
+   */
+  removeGlobalEventListeners() {
+    document.body.removeEventListener('wheel', this.onPointerWheelGlobal);
+    document.body.removeEventListener(
+      pointerEventName('pointerup'),
+      this.onPointerUpGlobal
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    const { visible, chartZoom } = this.props;
+
+    if (visible) {
+      const changed = (...names) => this.changed(names, prevProps, this.props);
+
+      if (
+        changed(
+          'visible',
+          'nodes',
+          'centralNode',
+          'linkedNodes',
+          'nodesActive',
+          'nodeSelected'
+        )
+      ) {
+        drawNodes.call(this);
+      }
+
+      if (changed('visible', 'chartZoom') && chartZoom.applied) {
+        drawViewport.call(this);
+      }
+
+      if (changed('visible', 'nodes', 'textLabels', 'chartSize')) {
+        this.zoomToFit();
+      }
+    }
+  }
+
+  /**
+   * Returns true if any of the given props are different between given objects.
+   * Only shallow changes are detected.
+   */
+  changed(props, objectA, objectB) {
+    return (
+      objectA && objectB && props.some(prop => objectA[prop] !== objectB[prop])
+    );
+  }
+
+  /**
+   * Create D3 element selectors
+   */
+  selectD3Elements() {
+    this.el = {
+      svg: select(this.svgRef.current),
+      wrapper: select(this.wrapperRef.current),
+      nodeGroup: select(this.nodesRef.current),
+      viewport: select(this.viewportRef.current)
+    };
+  }
+
+  /**
+   * Setup D3 zoom behaviour on component mount
+   */
+  initZoomBehaviour() {
+    // Create zoom behaviour
+    this.zoomBehaviour = zoom()
+      // Ignore all user input default behaviour
+      .filter(() => false)
+      // Transition using linear interpolation
+      .interpolate(interpolate)
+      // When zoom changes
+      .on('zoom', () => {
+        // Transform the <g> that wraps the map
+        this.el.wrapper.attr('transform', event.transform);
+      });
+
+    this.el.svg.call(this.zoomBehaviour);
+  }
+
+  /**
+   * Handle pointer enter
+   */
+  onPointerEnter = () => {
+    this.isPointerInside = true;
+  };
+
+  /**
+   * Handle pointer leave
+   */
+  onPointerLeave = () => {
+    this.isPointerInside = false;
+  };
+
+  /**
+   * Handle global pointer up
+   */
+  onPointerUpGlobal = () => {
+    this.isPointerDown = false;
+    this.isPointerInside = false;
+  };
+
+  /**
+   * Handle pointer down
+   * @param {Object} event Event object
+   */
+  onPointerDown = event => {
+    this.isPointerDown = true;
+    this.isPointerInside = true;
+
+    this.onPointerMove(event, true);
+  };
+
+  /**
+   * Handle pointer wheel
+   * @param {Object} event Event object
+   */
+  onPointerWheel = event => {
+    const { scale = 1 } = this.props.chartZoom;
+
+    // Change zoom based on wheel velocity
+    this.props.onUpdateChartZoom({
+      scale: scale - (event.deltaY || 0) * scale * 0.0025,
```

Comment:
Where does this number come from?

Reply:
I've added a name for this, but essentially just trial and error to get a reasonable speed.